common/error.hpp

```diff
@@ -56,6 +56,10 @@ class Error {
 	operator bool() const {
 		return static_cast<bool>(this->code);
 	}
+
+	std::string String() const {
+		return code.message() + ": " + message;
```

Comment:
Honestly I don't know what is best here. It is primarily inspired by how Golang renders errors, but our error representation is subtly different. If we consider an error like:

```
MakeError(ParseError, "Could not parse JSON");
```

this might be rendered as one of these:

```
1. Could not parse JSON: Parse error

2. Parse error: Could not parse JSON

3. Could not parse JSON[Parse error]

4. Parse error[Could not parse JSON]
```

This would render the second one, and the idea is that the thing you put in the message when creating the error is more specific than the error code, so it might make sense to render it in that order.

I think my vote would still be on number 2, both because it is the least deviation from the current client, and to me intuitively it looks most natural. But this is obviously a question of preference.

Reply:
Number 2 is nice, I agree with that. Let's just keep this format in mind when writing the error code strings and error messages.